build_tools/build/src/ci_gen/job.rs

```diff
@@ -300,9 +341,30 @@ impl JobArchetype for StandardLibraryTests {
         } else {
             "standard-library"
         };
-        let job_name = format!("Standard Library Tests ({graal_edition})");
+        let job_mode_name = self.mode_name();
+        let job_name = format!("Standard Library Tests ({graal_edition}) ({job_mode_name})");
         let run_command = format!("backend test {test_scope}");
         let run_steps_builder = RunStepsBuilder::new(run_command).customize(move |step| {
+            let cleanup_engine_distribution = Step {
+                run: Some(format!("rm -rf {}", built_distribution_directories(engine_launcher))),
+                shell: Some(Shell::Bash),
+                ..Default::default()
+            };
+
+            let download_engine_distribution =
+                step::download_engine_distribution(target, engine_launcher, graal_edition);
+
+            let unpack_engine_distribution = Step {
+                name: Some("Unpack Engine Distribution".into()),
+                run: Some(
+                    "tar -xvf built-distribution.tar
+rm built-distribution.tar
+"
+                    .into(),
+                ),
+                ..Default::default()
+            };
```

Comment:
I see that the `unpack_engine_distribution` step appears multiple times.

Could we extract it to a function? I think avoiding duplication here will be better for maintainability.